vault/mount.go

```diff
@@ -659,6 +659,40 @@ func (c *Core) mountInternal(ctx context.Context, entry *MountEntry, updateStora
 	return nil
 }
 
+// pluginTypeFromMountEntry attempts to find a pluginType associated with the
+// specified MountEntry. Returns consts.PluginTypeUnknown in the following cases:
+//
+// * Multiple types match (also returns an error)
+// * No types match (no error)
+func (c *Core) pluginTypeFromMountEntry(ctx context.Context, entry *MountEntry) (consts.PluginType, error) {
+	var pluginTypes []consts.PluginType
+	if c.builtinRegistry == nil || entry == nil {
+		return consts.PluginTypeUnknown, nil
+	}
+
+	pluginType, _ := consts.ParsePluginType(entry.Table)
+	if pluginType != consts.PluginTypeUnknown {
+		pluginTypes = append(pluginTypes, pluginType)
+	}
+
+	if c.builtinRegistry.Contains(entry.Type, consts.PluginTypeSecrets) {
+		pluginTypes = append(pluginTypes, consts.PluginTypeSecrets)
+	}
+
+	if c.builtinRegistry.Contains(entry.Type, consts.PluginTypeDatabase) {
+		pluginTypes = append(pluginTypes, consts.PluginTypeDatabase)
+	}
```

Comment:
These seem like unsafe assumptions to me. e.g. If we are processing an externally registered k8s auth plugin, it will get labelled as a secrets engine because there is also a kubernetes secret engine

Reply:
Agreed, and this actually makes for a good test case. I'll see about adding that to the PR.